### PR TITLE
Backport the fix from 9.2.2.

### PR DESCRIPTION
**Pull request by sellout**
Fixes a case where we would hang on single-core machines.

### CODE REVIEW

File: web/src/main/scala/quasar/server/Http4sUtils.scala [hunk 1]
```diff
@@ -45,13 +45,11 @@ object Http4sUtils {
...
-        .handle { case _: java.lang.InterruptedException => () }
-      test <- Task.delay(Option(System.console).isEmpty || System.in.available() <= 0)
-        .handle { case _ => true }
-      done <- if (test) waitForInput else Task.now(())
-    } yield done
+
+    val inputPresent = Task.delay(Option(System.console).nonEmpty && System.in.available() > 0)
+                           .handle { case _ => false }
+
+    Process.eval(inputPresent.after(250.milliseconds)).repeat.takeWhile(!_).run
```
Comment (mossprescott) [inline web/src/main/scala/quasar/server/Http4sUtils.scala]: Wow, this is an old version. This fixes a memory leak, so that's not a bad thing.

**Comment by mossprescott** [comment]
LGTM

File: scripts/publishJarOnUpstream [hunk 2]
```diff
@@ -14,8 +14,8 @@ if [[ ! -v GITHUB_TOKEN ]] ; then
     exit 0
 fi
 
-# only publish on quasar-analytics/quasar#master
-if [[ "$TRAVIS" == "true" && "$TRAVIS_BRANCH" == "master" && "$TRAVIS_REPO_SLUG" == "quasar-analytics/quasar" ]] ; then
+# only publish on quasar-analytics/quasar
+if [[ "$TRAVIS" == "true" && "$TRAVIS_REPO_SLUG" == "quasar-analytics/quasar" ]] ; then
```
Comment (mossprescott) [inline scripts/publishJarOnUpstream]: It's not obvious to me that this won't publish PRs, but I suppose we'll find out soon enough.

Comment (sellout) [inline scripts/publishJarOnUpstream]: Yeah … I tried to find the values for these variables in PR builds, but they don’t seem to be reported in the build log.

Comment (sellout) [inline scripts/publishJarOnUpstream]: Maybe we should name these branches like `backport/v4.5.6+`?

Comment (mossprescott) [inline scripts/publishJarOnUpstream]: Indeed, it looks like Travis tried to run the script and the chmod bits weren't right.

I think `backport/` is a good idea. Makes it clear that it's not being actively developed.

**Pull request closed without merging**